tempo_project/README.md

```diff
@@ -0,0 +1,398 @@
+
+### Working with Datasets
+
+In this repository, under `tempo_project/notebooks`, we provide example notebooks you can run using [pyspark](HERE), [duckdb](HERE), or [pandas](HERE) after you adjust `data_dir` to point to the location where you have downloaded one or more of the folders listed above and `dataset_name` to be the name of the folder containing the dataset you want to access.
+
+All three notebooks:
+- Load the `metadata.json` files to programmatically access the dataset column names, and use that information to discern, e.g., geographic resolution, whether the data are hourly or annual.
+- Display the initial timestamps (hourly data only).
+    - The timestamps stored in the dsgrid .parquet files are in UTC and cover the EST year 2012. Depending on exactly how the data are loaded, the starting timestamp might show up as `2012-01-01 00:00:00` (i.e., the starting timestamp in EST) or as `2012-01-01 05:00:00` (i.e., the starting timestamp in UTC).
+    - If the first timestamp is `2012-01-01 05:00:00` the notebook also demonstrates how to adjust the timestamps to start with `2012-01-01 00:00:00` as expected.
+- Validate the data by:
+    - Re-creating the lefthand side of Figure ES-1 in https://www.nrel.gov/docs/fy23osti/83916.pdf
+    - For hourly data only, plotting daily load profiles for two different days, one during standard time and one during daylight savings time, and for four different geographies, one for each time zone in the contiguous United States
+
+#### Loading the metadata
+
+Although the dsgrid query interface provides other options, all of the published datasets have at most one column per dimension type and have been written in unpivoted format. Thus, the column names can be loaded as in this example:
+
+```python
+import json
+from pathlib import Path
+
+def get_metadata(dataset_path):
+    with open(dataset_path / "metadata.json") as f:
+        result = json.load(f)
+    return result
+
+# load metadata and get column names by type
+metadata = get_metadata(data_dir / dataset_name) # data_dir is a pathlib.Path
+assert metadata["table_format"]["format_type"] == "unpivoted", metadata["table_format"]
+value_column = metadata["table_format"]["value_column"]
+columns_by_type = {dim_type: metadata["dimensions"][dim_type][0]["column_names"][0] for dim_type in metadata["dimensions"]}
+```
+
+Note that although a column name is provided for each dimension type, trivial dimensions (i.e., those with only one possible value, like `weather_year`) are not included in the data files. Thus, not all column names listed in the metadata will actually be present in loaded data frames.
+
+
+#### examples-spark.ipynb
+
+Dependencies:
+- python=3.10
```

Comment:
Any particular reason for 3.10? 3.11 is generally available and we currently do all our work on 3.12.